test/fixed_datetime_codec_test.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:convert/src/fixed_datetime_codec.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test('Parse only year', () {
+    var time = FixedDateTimeCodec('YYYY').decodeToLocal('1996');
+    expect(time, DateTime(1996));
+  });
+  test('Escaped chars are ignored', () {
+    var time = FixedDateTimeCodec('YYYY kiwi MM').decodeToLocal('1996 rnad 01');
+    expect(time, DateTime(1996, 1));
+  });
+  test('Parse two years throws', () {
+    expect(() => FixedDateTimeCodec('YYYY YYYY'), throwsException);
+  });
+  test('Parse year and century', () {
+    var time = FixedDateTimeCodec('CCYY').decodeToLocal('1996');
+    expect(time, DateTime(1996));
```

Comment:
Does this work (since `dt.year.toString()` is four digits and `YY` only allows two)?

Reply:
Good point, thanks!